src/main/java/com/eternalcode/combat/fight/pearl/FightPearlController.java

```diff
@@ -73,4 +77,47 @@ void onInteract(PlayerInteractEvent event) {
 
         this.fightPearlManager.markDelay(uniqueId);
     }
+
+    @EventHandler(priority = EventPriority.LOW)
+    void onEntityDamage(EntityDamageByEntityEvent event) {
+        if (!this.settings.pearlThrowControlEnabled) {
+            return;
+        }
+
+        if (!(event.getEntity() instanceof Player player)) {
+            return;
+        }
+
+        UUID playerUniqueId = player.getUniqueId();
+
+        if (!(event.getDamager() instanceof EnderPearl)) {
+            return;
+        }
+
+        if (event.getCause() != EntityDamageEvent.DamageCause.FALL) {
+            return;
+        }
+
+        if (!this.settings.pearlThrowMarksCombat) {
+            event.setCancelled(true);
```

Comment:
Rozumiem, co chcesz zrobić i że to działa. W pewnych warunkach modyfikowanie flow eventu i sztuczne przyznawanie damage graczowi może mieć potencjalne skutki uboczne typu:
~Gracz dostaje damage od pereł na chronionym regionie.
~System rankingu gildii nie uwzględnia odpowiednio tego damagr do asyst. 

Proponuję inne rozwiązanie. 
Dodaj listę zablokowanych entity które nie będą przyznawał taga. Np. ARROW, PEARL. Będzie to już raczej w tych głównych ficzerach.